### DISCUSSION TITLE
How to add a custom block which contains a table, buttons and pagination feature in GrapesJS?

### DISCUSSION THREAD

**Discussion by jithinvinod72** [GrapesJS/grapesjs#3536] [Q&A]
Hi all i recently started inventing Grapes Js and building sample projects using this.
I need to develop a cutom block which should give me a table its supporting buttons etc.
For example  , Lets say the Blocks name is 'Patients' and when i drag it to canvas it should give me a table with Patient Details along with an ADD PATIENT Button.
Can anybody tell how to do this?

**Comment by artf** [comment]
You have to use [Custom Components](https://grapesjs.com/docs/modules/Components.html#define-custom-component-type)
